src/main/java/seedu/task/logic/parser/ListParser.java

```diff
@@ -0,0 +1,33 @@
+package seedu.task.logic.parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import static seedu.taskcommons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import seedu.task.logic.commands.Command;
+import seedu.task.logic.commands.IncorrectCommand;
+import seedu.task.logic.commands.ListTaskCommand;
+
+public class ListParser implements Parser {
+
+	private static final Pattern LIST_ARGS_FORMAT = Pattern.compile("(?<type>-t|-e)" + "(?: (?<showAll>-a))*");
+	private static final String LIST_TYPE_TASK = "-t";
+
+	@Override
+	public Command prepare(String args) {
+		final Matcher matcher = LIST_ARGS_FORMAT.matcher(args.trim());
+
+		if (!matcher.matches()) {
+			return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListTaskCommand.MESSAGE_USAGE));
```

Comment:
For discussion: When we return the incorrect_command feedback message, should we display a specific example unique to list task or event only? Or should we show put MESSAGE_USAGE under the super ListCommand class with an example applicable to both task and event?
I am in favour of displaying both examples when incorrect command is keyed in